src/main/java/org/mozilla/zest/impl/HttpClientWrapper.java

```diff
@@ -0,0 +1,169 @@
+import org.apache.commons.httpclient.auth.AuthScope;
+import org.apache.commons.httpclient.methods.*;
+import org.apache.commons.httpclient.params.HttpClientParams;
+import org.mozilla.zest.core.v1.ZestAuthentication;
+import org.mozilla.zest.core.v1.ZestHttpAuthentication;
+import org.mozilla.zest.core.v1.ZestRequest;
+import org.mozilla.zest.core.v1.ZestResponse;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.Locale;
+
+public class HttpClientWrapper implements ZestHttpClient{
+
+    private HttpClient httpclient = new HttpClient();
+
+    public HttpClientWrapper(HttpClient httpclient) {
+        this.httpclient = httpclient;
+    }
+
+    public void setParams(HttpClientParams params) {
+        httpclient.setParams(params);
+    }
+
+    @Override
+    public void setAuthentication(ZestAuthentication zestAuthentication) {
+        if (zestAuthentication instanceof ZestHttpAuthentication) {
+            setHttpAuthentication((ZestHttpAuthentication) zestAuthentication);
+        }
+    }
+
+    @Override
+    public void setProxy(String host, int port) {
+        httpclient.getHostConfiguration().setProxy(host, port);
+    }
+
+    public void setHttpAuthentication(ZestHttpAuthentication zestHttpAuthentication) {
+        Credentials defaultCredentials = new UsernamePasswordCredentials(zestHttpAuthentication.getUsername(), zestHttpAuthentication.getPassword());
+        AuthScope authScope = new AuthScope(zestHttpAuthentication.getSite(), 80, AuthScope.ANY_REALM);
+        httpclient.getState().setCredentials(authScope, defaultCredentials);
+    }
+
+    @Override
+    public ZestResponse send(ZestRequest req)
+            throws IOException {
+        HttpMethod method;
+        URI uri = new URI(req.getUrl().toString(), false);
+
+        switch (req.getMethod()) {
+            case "GET":
+                method = new GetMethod(uri.toString());
+                // Can only redirect on GETs
+                method.setFollowRedirects(req.isFollowRedirects());
+                break;
+            case "POST":
+                method = new PostMethod(uri.toString());
+                break;
+            case "OPTIONS":
+                method = new OptionsMethod(uri.toString());
+                break;
+            case "HEAD":
+                method = new HeadMethod(uri.toString());
+                break;
+            case "PUT":
+                method = new PutMethod(uri.toString());
+                break;
+            case "DELETE":
+                method = new DeleteMethod(uri.toString());
+                break;
+            case "TRACE":
+                method = new TraceMethod(uri.toString());
+                break;
+            default:
+                throw new IllegalArgumentException("Method not supported: "
+                        + req.getMethod());
+        }
+
+        setHeaders(method, req.getHeaders());
+
+        for (Cookie cookie : req.getCookies()) {
```

Comment:
Probably better to use `getZestCookies()`, while it's easier to use `getCookies()` it will be removed sooner or later.